### PR TITLE
Recursively zip results folder using relative path

### PR DESCRIPTION
**Pull request by hnpl**
Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>

### CODE REVIEW

File: run/gem5art/run.py [hunk 1]
```diff
@@ -472,8 +472,11 @@ def saveResults(self) -> None:
 
         with zipfile.ZipFile(os.path.join(self.outdir, 'results.zip'), 'w',
                              zipfile.ZIP_DEFLATED) as zipf:
-            for f in files:
-                zipf.write(os.path.join(self.outdir, f))
+            for root, dirs, files in os.walk(self.relative_outdir):
+                if root == self.relative_outdir:
+                    files = filter(lambda f: f != 'results.zip', files)
+                for f in fiels:
```
Comment (powerjg) [inline run/gem5art/run.py]: typo! Someone didn't test this code ;)

File: run/gem5art/run.py [hunk 2]
```diff
@@ -472,8 +472,11 @@ def saveResults(self) -> None:
 
         with zipfile.ZipFile(os.path.join(self.outdir, 'results.zip'), 'w',
                              zipfile.ZIP_DEFLATED) as zipf:
-            for f in files:
-                zipf.write(os.path.join(self.outdir, f))
+            for root, dirs, files in os.walk(self.relative_outdir):
+                if root == self.relative_outdir:
```
Comment (powerjg) [inline run/gem5art/run.py]: This won't merge cleanly with the changeset removing the relative outdir... What will need to be done to fix this?

**Review by powerjg — COMMENTED** [review]
I can't find my other comment to delete it. Please ignore if you see it.

File: run/gem5art/run.py (hunk 2 above)
Comment (hnpl) [reply run/gem5art/run.py]: So when we create the gem5Run object, we must give `outdir` an abs path?

Comment (powerjg) [reply run/gem5art/run.py]: Yes. See https://github.com/darchr/gem5art/pull/50 and https://github.com/darchr/gem5art/pull/50/commits/9a139d613e8c11ab4ace0647ec26d75d05607a13.

I don't think there's any way around that. However, there will be a `cwd` in run (https://github.com/darchr/gem5art/pull/50/commits/9a139d613e8c11ab4ace0647ec26d75d05607a13#diff-7df16e1be50255dbdcbdaefa049d0cc3R379) if you wanted to use a directory relative to "results".

However, it's probably better not to zip based on the "results" directory, but to just zip the files in the output. So, if the output dir is `/home/jlp/gem5/experiment/results/linux/X86/cooltest/config1/` I would simply zip relative to `config1/`

Comment (hnpl) [reply run/gem5art/run.py]: So we can find the first subdirectory of that `cwd` that has a file, and zip everything relative to that folder?

Comment (hnpl) [reply run/gem5art/run.py]: Oh, we can simply look for the subdirectory containing simout / simerr files and zip everything relative to that folder.

**Comment by powerjg** [comment]
Yeah. That sounds right. However, you don't have to search. That directory
is outdir.

**Comment by powerjg** [comment]
What's the status of this? No rush, but I'd like to include this change in the next release as well.

**Comment by aakahlow** [comment]
I am going to look at this as Hoa is going to be busy with his class stuff today

**Review by powerjg — APPROVED** [review]
LGTM, assuming you've tested it :)

**Pull request merged**